sql/sql_const.h

```diff
@@ -29,7 +29,7 @@
 /* extra 4+4 bytes for slave tmp tables */
 #define MAX_DBKEY_LENGTH (NAME_LEN*2+1+1+4+4)
 #define MAX_ALIAS_NAME 256
-#define MAX_FIELD_NAME 34			/* Max colum name length +2 */
+#define MAX_FIELD_NAME (NAME_CHAR_LEN+2)			/* Max colum name length +2 */
```

Comment:
Why you used NAME_CHAR_LEN? I see that NAME_LEN is used in most places for an array allocation (one extra byte, plus one for zero termination). So i'd suggest NAME_LEN+1, as MAX_FIELD_NAME is used for an array size without +1.

Reply:
`NAME_CHAR_LEN` is the max length in *characters*, `NAME_LEN` is the max length in *bytes*.

`MAX_FIELD_NAME` is in bytes. So it must be `NAME_LEN`, `NAME_CHAR_LEN` is simply defined in different units.

Reply:
Agree, so it should be `(NAME_LEN + 1)`